ArithmeticTools/Rational.swift

```diff
@@ -100,16 +100,15 @@ public protocol Rational:
     init(_ numerator: Int, _ denominator: Int)
 
     /// - returns: Representation of a `Rational` value with a given `numerator`, if possible.
-    /// Otherwise, `nil`.
     ///
     /// > Preserves the arithmetic value of the original `Rational` value.
-    func respelling(numerator: Int) -> Self?
+    func respelling(numerator: Int) -> Self
 
     /// - returns: Representation of a `Rational` value with a given `denominator`, if
-    /// possible. Otherwise, `nil`.
+    /// possible. Assumes the denominator is not 0.
     ///
     /// > Preserves the arithmetic value of the original `Rational` value.
-    func respelling(denominator: Int) -> Self?
+    func respelling(denominator: Int) -> Self
```

Comment:
Also, is there a better word for this than `respelling` ? This was just an intuitive thing.